### PR TITLE
Add `unit` and `metric_type` meta to system fields.yml

### PR DESCRIPTION
**Pull request by exekias**
## What does this PR do?

This change updates `fields.yml` defintions to include units and metric
type information
## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/CONTRIBUTING.md#tips-for-building-integrations) and this pull request is aligned with them.
- [x] I have verified that all datasets collect metrics or logs.

## How to test this PR locally

Once https://github.com/elastic/kibana/issues/82273 is in, system package templates should include `unit` and `metric_type` _meta fields.

## Related issues

- Relates https://github.com/elastic/package-spec/issues/42

### CODE REVIEW

**Comment by ruflin** [comment]
Nice! Already lots of place where this is used.

**Review by fearful-symmetry — APPROVED** [review]
Super happy to see us finally adding this.

File: packages/system/data_stream/core/fields/fields.yml [hunk 1]
```diff
@@ -3,77 +3,102 @@
...
         The percentage of CPU time spent in user space.
     - name: user.ticks
       type: long
+      metric_type: counter
       description: |
         The amount of CPU time spent in user space.
     - name: system.pct
       type: scaled_float
       format: percent
+      unit: percent
```
Comment (jsoriano) [inline packages/system/data_stream/core/fields/fields.yml:24]: `unit` and `format` seem a bit redundant most of the times, maybe one can be derived from the other. Or maybe we could add additional checks, e.g. you probably always want to format as percent a metric with percent unit, same with bytes.

**Review by jsoriano — APPROVED** [review]
Nice!

File: packages/system/data_stream/core/fields/fields.yml [hunk 2]
```diff
@@ -3,77 +3,102 @@
   fields:
     - name: id
       type: long
+      metric_type: gauge
```
[the remark this answers is not in the document]
Comment (exekias) [reply packages/system/data_stream/core/fields/fields.yml:6]: good point, will change

File: packages/system/data_stream/core/fields/fields.yml (hunk 1 above)
Comment (exekias) [reply packages/system/data_stream/core/fields/fields.yml:24]: This sounds reasonable, I would prefer to keep them separated for now, as they end up in different places (template vs index pattern), I agree we could do better at linting.

**Comment by exekias** [comment]
Thanks for the reviews! @jsoriano this is ready for another go 😇

**Review by jsoriano — APPROVED** [review]

**Pull request merged**